classes/mail/variables/ReviewAssignmentEmailVariable.inc.php

```diff
@@ -15,21 +15,29 @@
 
 namespace PKP\mail\variables;
 
+use APP\core\Application;
+use PKP\context\Context;
+use APP\facades\Repo;
 use PKP\core\PKPApplication;
+use PKP\submission\PKPSubmission;
 use PKP\submission\reviewAssignment\ReviewAssignment;
 
 class ReviewAssignmentEmailVariable extends Variable
 {
     const REVIEW_DUE_DATE = 'reviewDueDate';
     const RESPONSE_DUE_DATE = 'responseDueDate';
-    const SUBMISSION_REVIEW_URL = 'submissionReviewUrl';
+    const REVIEW_ASSIGNMENT_URL = 'reviewAssignmentUrl';
 
-    /** @var ReviewAssignment $reviewAssignment */
-    protected $reviewAssignment;
+    protected ReviewAssignment $reviewAssignment;
+    protected PKPSubmission $submission;
+    protected Context $context;
 
     public function __construct(ReviewAssignment $reviewAssignment)
     {
         $this->reviewAssignment = $reviewAssignment;
+        $this->submission = Repo::submission()->get($this->reviewAssignment->getSubmissionId());
+        $contextDao = Application::getContextDAO();;
+        $this->context = $contextDao->getById($this->submission->getData('contextId'));
```

Comment:
This is fine for now, but eventually I'd like to figure out a way to avoid these database lookups within the email variables. They won't scale well if, for example, we were sending thousands of emails by the review assignment.

Maybe it's just something to leave for a caching layer that would solve a problem like this. Alternatively, we could consider passing in all other variables when an `EmailVariable` is instantiated in `Mailable::setupVariables()`.

But not something to try and solve now. It's not yet a performance issue for us.